CI mkdocs test case often fails

<!---
Verify first that your issue/request is not already reported on GitHub. -->

## Issue Type

<!--- Pick one below and delete the rest -->
- [X] Bug Report

## Summary

<!--- Explain the problem briefly -->
CI test "mkdocs" often fail during automated testing. Rerunning usually works. Sometimes it needs repeated attempts.

## Component Name

ansible-avd CI

## Steps to reproduce
<!--- For bugs, show exactly how to reproduce the problem, using a minimal test-case.
For new features, show how the feature would be used. -->

Commit to an open PR and wait for CI to run.

<!--- You can also paste gist.github.com links for larger files -->

### Expected results

<!--- What did you expect to happen when running the steps above? -->

mkdocs to work as long as I don't have typos.

### Actual results
<!--- What actually happened? If possible run with extra verbosity (-vvvv) -->

<!--- Paste verbatim command output between quotes below -->

mkdocs fail after the 30-second delay.

2 different failures have been identified:

- Unable to docker pull curl image

```bash
Unable to find image 'appropriate/curl:latest' locally
latest: Pulling from appropriate/curl
ff3a5c916c92: Pulling fs layer
...
58ae3cb4aac4: Pull complete
Digest: sha256:c8bf5bbec6397465a247c2bb3e589bb77e4f62ff88a027175ecb2d9e4f12c9d7
Status: Downloaded newer image for appropriate/curl:latest
Error: Process completed with exit code 7.
```

- Timeout to reach always the same destination: seems to be related to Actions infrastructure

```bash
http://127.0.0.1:8000/roles/eos_l3ls_evpn
	timeout	tools.ietf.org/html/rfc7432#section-8.3.1
```

__Related file:__ https://github.com/aristanetworks/ansible-avd/blob/devel/.github/workflows/documentation-check.yml

--- a/.github/workflows/documentation-check.yml
+++ b/.github/workflows/documentation-check.yml
@@ -29,7 +29,7 @@ jobs:
 
       - name: check links for 404
         run: |
-          docker run --network container:webdoc_avd raviqqe/muffet:1.5.7 http://127.0.0.1:8000/ -e ".*fonts.gstatic.com.*" -e ".*edit.*" -f --limit-redirections=3 --timeout=30
+          docker run --network container:webdoc_avd raviqqe/muffet:1.5.7 http://127.0.0.1:8000/ -e ".*fonts.gstatic.com.*" -e ".*tools.ietf.org.*" -e ".*edit.*" -f --limit-redirections=3 --timeout=30
 
       - name: 'stop docker-compose stack'
         run: |

--- a/.github/workflows/documentation-check.yml
+++ b/.github/workflows/documentation-check.yml
@@ -24,7 +24,7 @@ jobs:
       - name: 'test connectivity to mkdoc server'
         run: |
           sleep 30
-          docker run --network container:webdoc_avd appropriate/curl -s -I --retry 10 --retry-connrefused http://localhost:8000/
+          docker run --network container:webdoc_avd titom73/curl -s -I --retry 10 --retry-connrefused http://localhost:8000/
           docker ps
 
       - name: check links for 404

